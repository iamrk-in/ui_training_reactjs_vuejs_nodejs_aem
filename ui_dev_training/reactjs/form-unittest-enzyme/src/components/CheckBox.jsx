import React from "react";
import PropTypes from "prop-types";

const CheckBox = (props) => {
    return(
        <div className={props.className}>
            <input className="form-check-input" name={props.labelFor} type="checkbox" id={props.labelFor} onChange={props.handleChange} />
            <label className="form-check-label" htmlFor={props.labelFor}>{props.label}</label>
        </div>
    )
}

CheckBox.propTypes = {
    className: PropTypes.string, 
    lableFor: PropTypes.string, 
    checked: PropTypes.bool, 
    handleChange: PropTypes.func, 
    label: PropTypes.string
};

CheckBox.defaultProps = {
    className: 'form-group', 
    labelFor: 'checkbox', 
    checked: false, 
    handleChange: function(){}, 
    label: 'CheckBox'
};

export default CheckBox;