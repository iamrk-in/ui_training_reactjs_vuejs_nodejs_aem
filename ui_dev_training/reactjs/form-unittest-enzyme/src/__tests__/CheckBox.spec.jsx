import React from "react";
import renderer from "react-test-renderer";

import { mount } from "enzyme";
import enzymeConfig from "../../enzymeConfig";
import CheckBox from '../components/CheckBox';

describe("CheckBox", function() {
    
    it("should render correctly", function() {
        const tree = renderer.create(<CheckBox />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should have a single label tag', function() {
        const component = mount(<CheckBox />);
        expect(component.find('input').length).toBe(1);
    });

    it('should have the correct checkbox label', function() {
        var label = 'label';
        const component = mount(<CheckBox label={label} />);
        expect(component.find('label').text()).toEqual(label);
    });

    it('should call changeHandler once onChange', function() {
        const mockHandler = jest.fn();
        const component = mount(<CheckBox handleChange={mockHandler} />);
        const input = component.find('input').at(0);
        input.checked = true;
        input.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);
    });
   

});