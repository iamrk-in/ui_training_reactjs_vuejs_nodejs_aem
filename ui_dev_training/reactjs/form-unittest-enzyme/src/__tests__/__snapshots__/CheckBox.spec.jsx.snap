// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should render correctly 1`] = `
<div
  className="form-group"
>
  <input
    className="form-check-input"
    id="checkbox"
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="form-check-label"
    htmlFor="checkbox"
  >
    CheckBox
  </label>
</div>
`;
